repos/system_upgrade/el7toel8/actors/setuprhsmonrhel8/actor.py

```diff
@@ -0,0 +1,28 @@
+from leapp.actors import Actor
+from leapp.libraries.actor import library
+from leapp.models import UsedTargetRepositories, TargetRHSMInfo
+from leapp.tags import FirstBootPhaseTag, IPUWorkflowTag
+
+
+class SetUpRHSMOnRHEL8(Actor):
+    """
+    On the upgraded RHEL 8, set release and enable repositories that were used during the upgrade transaction.
+
+    We want users to have all the RHEL 8 repos that are equivalent to the enabled RHEL 7 ones available on the upgraded
+    system. The BaseOS and AppStream repos are enabled on RHEL 8 by default. Any other repository needs to be enabled
+    specifically using the Red Hat Subscription Manager (RHSM) utility.
+
+    We need to overwrite any RHSM release that may have been set before the upgrade, e.g. 7.6. Reasons:
+    - If we leave the old RHEL 7 release set, dnf calls on the upgraded RHEL 8 would fail.
+    - If we merely unset the release, users might end up updating the system to a newer version than the upgrade
+      was supposed to be upgrading to.
+    """
+
+    name = 'set_up_rhsm_on_rhel8'
+    consumes = (UsedTargetRepositories, TargetRHSMInfo)
+    produces = ()
+    tags = (IPUWorkflowTag, FirstBootPhaseTag)
+
+    def process(self):
+        library.set_release()
+        library.enable_repos()
```

Comment:
This will crash for custom repos.

Reply:
I'll issue a warning in case of the use of custom repos unknown to submgr.